server/public/model/cluster_info.go

```diff
@@ -4,9 +4,10 @@
 package model
 
 type ClusterInfo struct {
-	Id         string `json:"id"`
-	Version    string `json:"version"`
-	ConfigHash string `json:"config_hash"`
-	IPAddress  string `json:"ipaddress"`
-	Hostname   string `json:"hostname"`
+	Id            string `json:"id"`
+	Version       string `json:"version"`
+	SchemaVersion string `json:"schema_version"`
```

Comment:
> How is SchemaVersion different than Version in practice?

Basically what Ibrahim said. But also schema version is always numeric and does not go into the various flavors of MM version like master, vXX-RC, then dot releases etc. It's a simple way to directly check the latest schema installed.

>  But it's trickier if <ESR-1 comes online later and has to figure out if it's allowed to join the cluster.

Right, I believe that we agreed to periodically check for this and simply restart the job if we find an older node suddenly joining the cluster.

Overall Jesse, this PR is targeted towards solving "when do you start a background migration job?" to unblock some new migrations that Ibrahim is planning. 

The ESR compatiblity problem is definitely something we want to tackle, but I would say this is a simple first step that will allow us to start adding "ALTER TABLE ADD COLUMN" migrations which require data backfilling after the new column is added. Basically following this [example](https://docs.google.com/document/d/1wdQLzS_C6cjJIcyZWEYEeaP3Ej_NuuXj4lDt1VFuyZU/edit#heading=h.f6dz13s1pdbr).

Reply:
Ah, thank you for clarifying. I better understand that this isn't focused on the ESR compatibility case, but (as the PR already summarizes ;P) the background job migration case.
 
> Two different versions of mattermost can have the same schema version if there are no migrations added.

I'm still wondering why we don't just use the existing `Version` for this, as it's surely a stronger guarantee than comparing `SchemaVersion`. If by chance we have to patch and restart the job in a dot release, we won't be able to key off `SchemaVersion` to know when it's safe to start over. It seems simpler just to require a uniform cluster (both `Version` and implicitly `SchemaVersion`).

Reply:
> Good point. I agree. It would be safer to compare both and ensure everything is on the same version.

To clarify, comparing `Version` is sufficient to compare both, right? (How could the same version of Mattermost have a different `SchemaVersion`?)

Reply:
In rare cases, it can be different. For example, in an environment of daily deployments like Community, the version is `master.<number>`. Sometimes due to glitches in Gitlab CI, we have seen that version number to be incorrect and the same across multiple commits.

Lastly, the version in dev environments is `dev`.

So having a separate check helps to avoid weird Gitlab bugs and also allow to easily test upgrade scenarios locally without having to modify any code to bump up the version manually.

Reply:
Sounds good, thank you for building a case for the extra "complexity" :)